bemani/backend/iidx/rootage.py

```diff
@@ -0,0 +1,1299 @@
+# vim: set fileencoding=utf-8
+import copy
+import struct
+from typing import Optional, Dict, Any
+
+from bemani.backend.iidx.base import IIDXBase
+from bemani.backend.iidx.sinobuz import IIDXSinobuz
+from bemani.backend.iidx.cannonballers import IIDXCannonBallers
+
+from bemani.common import ValidatedDict, Model, VersionConstants, Time, ID, intish
+from bemani.data import Data, UserID
+from bemani.protocol import Node
+
+# todo: clear '?' mark
+class IIDXRootage(IIDXBase):
+
+    name = 'Beatmania IIDX ROOTAGE'
+    version = VersionConstants.IIDX_ROOTAGE
+    sinobuz = None

```

Comment:
Why do we need a handle specifically to sinobuz? If you need old version support, you should use the `previous_version` paradigm that's used across the board.

Reply:
Yes it's an instance that proxied for avoding duplicate codes. I'm agreed with you that we need to reduce the coupling of two modules.